### PR TITLE
fix: changed redhatsso to return same errors as legacy sso

### PR DESCRIPTION
**Pull request by ziccardi**
### Error deleting non-existent service account
```
curl -v -X DELETE -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/kafkas_mgmt/v1/service_accounts/7d7426a8-9f03-408b-880f-997086b09cc3
```
#### REDHAT-SSO
```
< HTTP/1.1 404 Not Found
< Content-Type: application/json
< Vary: Authorization
< X-Operation-Id: caophai87d5n6qdrle2g
< Date: Tue, 21 Jun 2022 10:08:48 GMT
< Content-Length: 205
< 
{"id":"113","kind":"Error","href":"/api/kafkas_mgmt/v1/errors/113","code":"KAFKAS-MGMT-113","reason":"service account not found 7d7426a8-9f03-408b-880f-997086b09cc3","operation_id":"caophai87d5n6qdrle2g"}
```

#### MAS-SSO
```
< HTTP/1.1 404 Not Found
< Content-Type: application/json
< Vary: Authorization
< X-Operation-Id: caopi9a87d5n9gbkial0
< Date: Tue, 21 Jun 2022 10:10:45 GMT
< Content-Length: 205
<
{"id":"113","kind":"Error","href":"/api/kafkas_mgmt/v1/errors/113","code":"KAFKAS-MGMT-113","reason":"service account not found 7d7426a8-9f03-408b-880f-997086b09cc3","operation_id":"caopi9a87d5n9gbkial0"}
```

### Error getting non-existent service account
```
curl -v -X GET -H "Authorization: Bearer $(ocm token)" http://localhost:8000/api/kafkas_mgmt/v1/service_accounts/7d7426a8-9f03-408b-880f-997086b09cc3
```
#### REDHAT-SSO
```
< HTTP/1.1 404 Not Found
< Content-Type: application/json
< Vary: Authorization
< X-Operation-Id: caopkqq87d5nabnm63r0
< Date: Tue, 21 Jun 2022 10:16:11 GMT
< Content-Length: 214
<
{"id":"113","kind":"Error","href":"/api/kafkas_mgmt/v1/errors/113","code":"KAFKAS-MGMT-113","reason":"service account not found clientId 7d7426a8-9f03-408b-880f-997086b09cc3","operation_id":"caopkqq87d5nabnm63r0"}
```

#### MAS-SSO
```
< HTTP/1.1 404 Not Found
< Content-Type: application/json
< Vary: Authorization
< X-Operation-Id: caopjsq87d5n9vasvc60
< Date: Tue, 21 Jun 2022 10:14:29 GMT
< Content-Length: 205
<
{"id":"113","kind":"Error","href":"/api/kafkas_mgmt/v1/errors/113","code":"KAFKAS-MGMT-113","reason":"service account not found 7d7426a8-9f03-408b-880f-997086b09cc3","operation_id":"caopjsq87d5n9vasvc60"}
```

### Error maximum number of service accounts reached
```
curl --location --request POST 'http://localhost:8000/api/kafkas_mgmt/v1/service_accounts' \
--header 'Authorization: Bearer '$TOKEN \
--header 'Content-Type: application/json' \
--data-raw '{
  "name": "testak'$i'",
  "description": "test'$i'"
}
'
```
```
{"id":"115","kind":"Error","href":"/api/kafkas_mgmt/v1/errors/115","code":"KAFKAS-MGMT-115","reason":"Max allowed number:50 of service accounts for user has reached","operation_id":"caorgpq87d5o8ohmatq0"}
```

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

**Comment by akoserwal** [comment]
Please can you update the service account integration test for redhat-sso

File: pkg/services/sso/redhatsso_service.go [hunk 1]
```diff
@@ -29,7 +69,7 @@ func (r *redhatssoService) DeRegisterClientInSSO(accessToken string, clientId st
 	glog.V(5).Infof("Deregistering client with id: %s", clientId)
 	err := r.client.DeleteServiceAccount(accessToken, clientId)
 	if err != nil {
-		return errors.NewWithCause(errors.ErrorFailedToDeleteSSOClient, err, "failed to delete the sso client")
+		return errors.NewWithCause(errors.ErrorFailedToDeleteSSOClient, err, getErrorDescription(err, "failed to delete the sso client"))
 	}
 	glog.V(5).Infof("Deregistered client with id: %s", clientId)
```
Comment (akoserwal) [inline pkg/services/sso/redhatsso_service.go:74]: getErrorDescription can also be added in the logs this will help to debug issues

File: pkg/services/sso/redhatsso_service.go [hunk 2]
```diff
@@ -29,7 +69,7 @@ func (r *redhatssoService) DeRegisterClientInSSO(accessToken string, clientId st
 	glog.V(5).Infof("Deregistering client with id: %s", clientId)
 	err := r.client.DeleteServiceAccount(accessToken, clientId)
 	if err != nil {
-		return errors.NewWithCause(errors.ErrorFailedToDeleteSSOClient, err, "failed to delete the sso client")
+		return errors.NewWithCause(errors.ErrorFailedToDeleteSSOClient, err, getErrorDescription(err, "failed to delete the sso client"))
```
Comment (akoserwal) [inline pkg/services/sso/redhatsso_service.go:72]: if an error description returned from sso can be logged as well. it will help in debugging issues.

File: pkg/services/sso/redhatsso_service.go [hunk 3]
```diff
@@ -127,9 +202,9 @@ func (r *redhatssoService) DeRegisterKasFleetshardOperatorServiceAccount(accessT
 		}
 	}
 
-	err := r.client.DeleteServiceAccount(accessToken, agentClusterId)
+	err := r.DeleteServiceAccount(accessToken, context.Background(), agentClusterId)
```
Comment (akoserwal) [inline pkg/services/sso/redhatsso_service.go:205]: why this change?

**Comment by akoserwal** [comment]
Left some minor comments, rest looks good to me

File: pkg/services/sso/redhatsso_service.go (hunk 3 above)
Comment (ziccardi) [reply pkg/services/sso/redhatsso_service.go:205]: Because this way I reuse the code in `DeleteServiceAccount` to retrieve the error message from the error returned by `redhat_sso`.

Comment (akoserwal) [reply pkg/services/sso/redhatsso_service.go:205]: make sense

**Comment by ziccardi** [comment]
@akoserwal Added the logs as requested!

**Review by akoserwal — APPROVED** [review]
